.github/drafts/16296/16296_self-hosting.md

```diff
@@ -0,0 +1,74 @@
+
+Appsmith provides a [cloud-hosted version](https://app.appsmith.com)[cloud-hosted] and the option to set up your own instance on some popular [platforms](# Deploy).
+
+## Cloud-hosted
+Cloud-hosted version is available over the internet, which makes it easy for users to sign up and start building apps. It's a perfect solution for users who don't have their own infrastructure setup. You can sign up on appsmith.com and follow the [Quick start guide](start-building) to get started.
+
+## Self-hosted
+You can choose to deploy Appsmith on your own infrastructure. It's often useful when you would want to plug it into existing infrastructure without the need to maintain a different setup. It enables you to take advantage of what your infrastructure has to offer and extend it to Appsmith. 
+
+ > Appsmith collects [usage data](Link to Telemetry) to serve you better. You can choose to [opt-out](Disable Telemetry from Admin Settings) at any time.
+
+Deploying Appsmith on your infrastructure offers the following benefits:
+
+#### Security
+You get to keep your data on-premise. It also ensures that your API keys or security credentials would never leave your server and have the same security rules applied to any other app. The applications you build would be on your infrastructure, inside your corporate networks. You may choose to host it on internal networks behind a firewall for a defined set of users. You may also configure your [custom domain](Link to custom domain page) and extend the security policies to Appsmith installation.
+
+Appsmith provides a [secured environment](Link to security page) for cloud-hosted and self-hosted instances by encrypting sensitive information like database credentials using [AES 256 Encryption](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard). For self-hosted instances, it also supports setting up [SSL certificates](Link to SSL & Custom Domain page - Setting up SSL section) via [LetsEncrypt](https://letsencrypt.org/) during installation.
+
+#### Internal database
+A self-hosted instance could allow you to integrate with an internal database and use it to build apps.
+
+> Using a cloud-hosted instance would require the ability to access the internal databases. You would be required to use tunneling services like [ngrok](Link to Ngrok doc on Appsmtih) to expose your internal database to the internet and build apps.
+
+Appsmith uses MongoDB for storing app data and Redis for storing session information. You may wish to integrate with custom MongoDB or Redis servers hosted on your infrastructure. For this, you would have to [modify the configuration of environment variables and point them to the custom MongoDB/Redis] (Link to Custom MongoDB and Redis Page) instances.
```

Comment:
We've started using Redis for non-critical caches as well, very recently.

Reply:
I'll add that.